packages/klighd-core/src/bookmarks/bookmark-panel.tsx

```diff
@@ -0,0 +1,71 @@
+/** @jsx html */
+import { html } from "snabbdom-jsx"; // eslint-disable-line @typescript-eslint/no-unused-vars
+import { inject, injectable, postConstruct } from "inversify";
+import { VNode } from "snabbdom/vnode";
+import { DISymbol } from "../di.symbols";
+import { RenderOptionsRegistry } from "../options/render-options-registry";
+import { PreferencesRegistry } from "../preferences-registry";
+import { SidebarPanel } from "../sidebar";
+import { SynthesesRegistry } from "../syntheses/syntheses-registry";
+import { TYPES, IActionDispatcher, Action } from "sprotty";
+
+@injectable()
+export class BookmarkPanel extends SidebarPanel {
+    
+    @inject(TYPES.IActionDispatcher) private actionDispatcher: IActionDispatcher;
+    @inject(DISymbol.SynthesesRegistry) private synthesesRegistry: SynthesesRegistry;
+    @inject(DISymbol.PreferencesRegistry) private preferencesRegistry: PreferencesRegistry;
+    @inject(DISymbol.RenderOptionsRegistry) private renderOptionsRegistry: RenderOptionsRegistry;
+
+    private static bookmarkActions: [key: string, icon: VNode, action: Action][]
```

Comment:
Instead of storing they state in a panel, it should be separated into a `Registry`. Separation between UI and Data. The Registry might be able handle the CreateBookmarkAction directly.

Reply:
added a BookmarkRegistry, but I don't think the Registry can handle the Action directly as it need access to the model which I don't know how to access from the Action in thee registry

Reply:
@Skgland My thought was that the command might dispatch an extra add action with the model attached. However, I am not sure whether or not a command is supposed to trigger further actions. The Sprotty architecture overview suggests otherwise (?). The way you handled it in the new commit is properly the best call.